node/src/neighborhood/mod.rs

```diff
@@ -4778,99 +4703,93 @@ mod tests {
     /*
             Database:
 
-            P---q---R
-
-            Test is written from the standpoint of P. Node q is non-routing.
+            root_key---c_au---b_fr
+                        |
+                       a_fr
+            Test is written from the standpoint of root_key.
     */
 
     #[test]
-    fn find_best_segment_traces_unreachable_country_code_exit_node() {
-        init_test_logging();
+    fn exit_node_not_found_due_to_country_code_strict_requirement() {
         let mut subject = make_standard_subject();
         let (recipient, _) = make_node_to_ui_recipient();
         subject.node_to_ui_recipient_opt = Some(recipient);
         subject.user_exit_preferences.fallback_preference =
             FallbackPreference::ExitCountryWithFallback;
         let message = UiSetExitLocationRequest {
             fallback_routing: false,
-            exit_locations: vec![CountryCodes {
+            exit_locations: vec![CountryGroups {
                 country_codes: vec!["CZ".to_string()],
                 priority: 1,
             }],
             show_countries: false,
         };
         let db = &mut subject.neighborhood_database;
-        let p = &db.root_mut().public_key().clone();
-        let a = &db.add_node(make_node_record(2345, true)).unwrap();
-        let b = &db.add_node(make_node_record(5678, true)).unwrap();
-        let c = &db.add_node(make_node_record(1234, true)).unwrap();
-        db.add_arbitrary_full_neighbor(p, c);
-        db.add_arbitrary_full_neighbor(c, b);
-        db.add_arbitrary_full_neighbor(c, a);
+        let root_key = &db.root_mut().public_key().clone();
```

Comment:
Vzhledem k tomu, ze u tech jinych take neuvadime, ze to jsou klice, tak ani tady bys nemusel pouzivat tu priponu `key`. Ale necham to na tobe.